src/geom/SpanSet.cc

```diff
@@ -0,0 +1,828 @@
+
+std::shared_ptr<geom::SpanSet> geom::SpanSet::findEdgePixels() const {
+    image::Mask<image::MaskPixel> tempMask(getBBox());
+    setMask(tempMask, static_cast<image::MaskPixel>(1));
+    auto erodedSpanSet = erode(1, geom::Stencil::CIRCLE);
+    erodedSpanSet->clearMask(tempMask, static_cast<image::MaskPixel>(1));
+    return geom::maskToSpanSet(tempMask);
+}
+
+std::shared_ptr<geom::SpanSet> geom::SpanSet::shiftedBy(int x, int y) const {
+    // Function to create a new SpanSet which is a copy of this, shifted by x and y
+    return makeShift(x, y);
+}
+
+std::shared_ptr<geom::SpanSet> geom::SpanSet::shiftedBy(geom::Extent2I const & offset) const {
+    // Function to create a new SpanSet which is a copy of this, shifted by an extent object
+    return makeShift(offset.getX(), offset.getY());
+}
+
+std::shared_ptr<geom::SpanSet> geom::SpanSet::makeShift(int x, int y) const {
+    // Implementation method common to all overloads of the shiftedBy method
+    std::vector<Span> tempVec;
+    tempVec.reserve(_spanVector.size());
+    for (auto const & spn : _spanVector) {
+        tempVec.push_back(Span(spn.getY() + y, spn.getMinX() + x, spn.getMaxX() + x));
+    }
+    return std::make_shared<geom::SpanSet>(std::move(tempVec), false);
+}
+
+std::shared_ptr<geom::SpanSet> geom::SpanSet::clippedTo(geom::Box2I const & box) const {
+    /* Return a copy of the current SpanSet but only with values which are contained within
+     * the supplied box
+     */
+    std::vector<Span> tempVec;
+    for (auto const & spn : _spanVector) {
+        if (spn.getY() >= box.getMinY() && spn.getY() <= box.getMaxY() &&
+            spansOverlap(spn, geom::Span(spn.getY(), box.getMinX(), box.getMaxX()))) {
+            tempVec.push_back(Span(spn.getY(), std::max(box.getMinX(), spn.getMinX()),
+                                   std::min(box.getMaxX(), spn.getMaxX())));
+        }
+    }
+    return std::make_shared<geom::SpanSet>(std::move(tempVec), false);
+}
+
+bool geom::SpanSet::overlaps(SpanSet const & other) const {
+    // Function to check if two SpanSets overlap
+    for (auto const & otherSpan : other) {
+        for (auto const & spn : _spanVector) {
+            if (spansOverlap(otherSpan, spn)) {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+bool geom::SpanSet::contains(geom::SpanSet const & other) const {
+    // Handle null SpanSet passed as other
+    if (other.size() == 0) {
+        return false;
+    }
+    // Function to check if a SpanSet is entirely contained within this
+    for (auto const & otherSpn : other) {
+        std::size_t counter = 0;
+        for (auto const & spn : _spanVector) {
+            // Check that the end points of the span from other are contained in the
+            // span from this
+            if (spn.contains(geom::Point2I(otherSpn.getMinX(), otherSpn.getY())) &&
+                spn.contains(geom::Point2I(otherSpn.getMaxX(), otherSpn.getY())))
+                ++counter;
```

Comment:
Multi-line `if` statements should use braces, even if the body is only one line.